tests/director/tests/bundle_instance_auth_per_scenario_test.go

```diff
@@ -0,0 +1,290 @@
+	})
+}
+
+func TestUnassignResourceFromScenarioUsingAutomaticScenarioAssignment(t *testing.T) {
+	ctx := context.Background()
+	tenantID := tenant.TestTenants.GetDefaultTenantID()
+	createScenarioLabelDefIfNotExist(t, ctx)
+
+	var cleanups []func()
+	testCleanUp := func() {
+		for _, cleanup := range cleanups {
+			cleanup()
+		}
+	}
+	defer testCleanUp()
+
+	commonScenario := "common-scenario"
+	otherScenario := "other-common-scenario"
+
+	scenarios := []string{conf.DefaultScenario, commonScenario, otherScenario}
+	fixtures.UpdateScenariosLabelDefinitionWithinTenant(t, ctx, dexGraphQLClient, tenant.TestTenants.GetDefaultTenantID(), scenarios)
+	cleanups = append([]func(){func() {
+		fixtures.UpdateScenariosLabelDefinitionWithinTenant(t, ctx, dexGraphQLClient, tenant.TestTenants.GetDefaultTenantID(), scenarios[:1])
+	}}, cleanups...)
+
+	// GIVEN
+	data, cleanup, ok := setupAppAndRuntimeWithCommonScenario(t, ctx, commonScenario)
+	cleanups = append([]func(){func() { cleanup() }}, cleanups...)
+	require.True(t, ok)
+
+	// Label selector for automatic scenario assignment
+	labelSelector := graphql.LabelSelectorInput{
+		Key:   "keyA",
+		Value: "valueA",
+	}
+
+	// Update application scenarios
+	setAppLabelRequest := fixtures.FixSetApplicationLabelRequest(data.appId, ScenariosLabel, []interface{}{commonScenario, otherScenario})
+	appLabel := graphql.Label{}
+	err := testctx.Tc.RunOperation(ctx, dexGraphQLClient, setAppLabelRequest, &appLabel)
+	require.NoError(t, err)
+
+	// Set runtime label
+	setRuntimeLabelRequest := fixtures.FixSetRuntimeLabelRequest(data.runtimeId, labelSelector.Key, labelSelector.Value)
+	runtimeLabel := graphql.Label{}
+	err = testctx.Tc.RunOperation(ctx, dexGraphQLClient, setRuntimeLabelRequest, &runtimeLabel)
+	require.NoError(t, err)
+
+	// Create automatic scenario assignment
+	asaInput := graphql.AutomaticScenarioAssignmentSetInput{
+		ScenarioName: otherScenario,
+		Selector:     &labelSelector,
+	}
+	fixtures.CreateAutomaticScenarioAssignmentInTenant(t, ctx, dexGraphQLClient, asaInput, tenantID)
+
+	// Cannot delete automatic scenario assignment when there are existing bundle instance auths
+	assignment := graphql.AutomaticScenarioAssignment{}
+	req := fixtures.FixDeleteAutomaticScenarioAssignmentForScenarioRequest(otherScenario)
+	err = testctx.Tc.RunOperationWithCustomTenant(ctx, dexGraphQLClient, tenantID, req, &assignment)
+	assert.Error(t, err)
+
+	// Cannot delete label which corresponds for the automatic scenario assignment
+	request := fixtures.FixDeleteRuntimeLabelRequest(data.runtimeId, labelSelector.Key)
+	label := graphql.Label{}
+	err = testctx.Tc.RunOperation(ctx, dexGraphQLClient, request, &label)
+	assert.Error(t, err)
+
+	// Deleting scenario for which there is automatic scenario assignment is successful
+	updateInput := graphql.RuntimeInput{
+		Name:   data.runtimeName,
+		Labels: graphql.Labels{ScenariosLabel: []string{commonScenario}, labelSelector.Key: labelSelector.Value},
+	}
```

Comment:
Why is this allowed? Is the label automatically assigned in the background and this is why it is not returned as error?

If that is the case we should add some more asserts to verify this behaviour.

Reply:
Yes, we are expecting the same error here. What do you mean by more asserts? The only one that comes in my mind is that if we remove the ASA selector label from the runtime and provide the matched scenario explicitly then this should be allowed to proceed.